pwndbg/commands/context.py

```diff
@@ -278,21 +279,23 @@ def context_backtrace(frame_count=10, with_banner=True):
     return result
 
 
-def context_args():
+def context_args(with_banner=True):
     result = []
 
-    ##################################################
-    # DISABLED FOR NOW, I LIKE INLINE DISPLAY BETTER
-    ##################################################
-    # # For call instructions, attempt to resolve the target and
-    # # determine the number of arguments.
-    # for arg, value in pwndbg.arguments.arguments(pwndbg.disasm.one()):
-    #     code   = False if arg.type == 'char' else True
-    #     pretty = pwndbg.chain.format(value, code=code)
-    #     result.append('%-10s %s' % (arg.name+':', pretty))
-    # if not result:
-    #         return []
-    # result.insert(0, pwndbg.ui.banner("arguments"))
+    # For call instructions, attempt to resolve the target and
+    # determine the number of arguments.
+    one = pwndbg.disasm.one()
+    for arg, value in pwndbg.arguments.get(one):
+        code   = False if arg.type == 'char' else True
+        pretty = pwndbg.chain.format(value, code=code)
+        result.append('%8s%-10s %s' % ('', N.argument(arg.name) + ':', pretty))
+
+    if not result:
+            return []
+
+    if with_banner:
+        result.insert(0, pwndbg.ui.banner("arguments"))
```

Comment:
I feel like we should move it to the bottom of the function as `if not with_banner: return []`

Reply:
the point is to only insert the banner if we want ti print something and the banner was enabled. We cant early exit if no banner is set. it would need more "useless" code and temp local vars to be able to check if we have results other then the banner. IMO for this case its less code if we keep as is